wagtail/core/models.py

```diff
@@ -1225,10 +1238,17 @@ def get_specific(self, deferred=False, copy_attrs=None):
             # Fetch object from database
             specific_obj = model_class._default_manager.get(id=self.id)
 
-        # Copy additional attribute values
-        for attr in copy_attrs or ():
-            if attr in self.__dict__:
+        # Copy non-field attribute values
+        if copy_attrs is not None:
+            for attr in (attr for attr in copy_attrs if attr in self.__dict__):
                 setattr(specific_obj, attr, getattr(self, attr))
+        else:
+            exclude = copy_attrs_exclude or ()
```

Comment:
We don't exclude the fields attributes here, so we're essentially resetting all the fields, so I'm not sure this "non-field" comment is accurate, as it's copying the fields too (unless I'm missing something)

Reply:
If we wanted to prevent setting them again, we could always exclude anything in `._meta.get_fields()`, but I don't suppose there's much overhead from setting them

Reply:
Yes, I tried a few different versions of this. In the end, it seemed clearest to avoid too many conditionals and go with `setdefault` for the setting, which would naturally avoid resetting anything unnecessary. I've added a comment in to help make this clearer.